src/StartPage/DifficultySelection.js

```diff
@@ -0,0 +1,122 @@
+        background-color: #fff;
+        color: #0094da;
+      }
+`;
+
+const StyledDiv = styled.div`
+    font-size: 22px;
+    padding-left: 50px 50px 50px 50px;
+    box-sizing: border-box;
+    margin-top: 20px;
+`;
+
+class DifficultySelection extends Component {
+    constructor(props){
+        super();
+        this.state = {
+            selectedOption: 'easy',
+            data: null,
+            loading: null
+        }
+    }
+
+    dataHandler = () => {
+        this.setState({ loading: true });
+        const selectedOption = this.state.selectedOption;
+        console.log(selectedOption);
+        const url = 'https://opentdb.com/api.php?amount=10&category=18&difficulty='+ selectedOption + '&type=multiple';
+        fetch(url)
+            .then(res => res.json())
+            .then(json => this.setState({ data: json,
+                                        loading: false}))
+            .catch((error) => {
+                console.error('Error:', error);
+            });
+    }
+
+    componentDidUpdate = (prevState) => {
+        if (this.state.loading === false){
+            const results = [...this.state.data.results];
+            this.props.onOptionSelected(results);
+        };
+    }
+
+    optionChangeHandler = (event) => {
+        this.setState({
+            selectedOption: event.target.value
+        });
+    }
+
+    render() {
+        return (
+            <StyledDiv>
+                Please select the level of questions:
+                <div>
+                    <label>
+                        <input 
+                            type="radio" 
+                            value="easy" 
+                            checked={this.state.selectedOption === 'easy'}
+                            onChange={this.optionChangeHandler} />
+                        Easy
+                    </label>
+                </div>
+                <div>
+                    <label>
+                        <input 
+                            type="radio" 
+                            value="medium"
+                            checked={this.state.selectedOption === 'medium'}
+                            onChange={this.optionChangeHandler} />
+                        Medium
+                    </label>
+                </div>
+                <div>
+                    <label>
+                        <input
+                            type="radio"
+                            value="hard"
+                            checked={this.state.selectedOption === 'hard'}
+                            onChange={this.optionChangeHandler}/>
+                        Hard
+                    </label>
+                </div>
+                <Fragment>
+                    <ClipLoader
+                        sizeUnit={"px"}
+                        size={50}
+                        color={'#0094da'}
+                        loading={this.state.loading}
+                    />
+                    {this.state.loading !== true && 
```

Comment:
this could be just !this.state.loading, or after deconstruction !loading, boolean variables do no need to be equation checked